downstream.go

```diff
@@ -546,6 +546,14 @@ func unmarshalUsername(rawUsername string) (username, client, network string) {
 	return username, client, network
 }
 
+func (dc *downstreamConn) unmarshalNicks(uc *upstreamConn, text string) string {
```

Comment:
Can we rename this to e.g. `unmarshalText`? `unmarshalNicks` doesn't make it clear this accepts an arbitrary string.